src/globus_cli/login_manager/tokenstore.py

```diff
@@ -10,15 +11,218 @@
 from .scopes import CURRENT_SCOPE_CONTRACT_VERSION
 
 if t.TYPE_CHECKING:
-    from globus_sdk.tokenstorage import SQLiteAdapter
+    from globus_sdk.tokenstorage import SQLiteAdapter  # noqa: F401
 
 # env vars used throughout this module
 GLOBUS_ENV = os.environ.get("GLOBUS_SDK_ENVIRONMENT")
 
 
-# stub to allow type casting of a function to an object with an attribute
-class _TokenStoreFuncProto:
-    _instance: SQLiteAdapter
+class CLITokenstorage:
+    """
+    A wrapper over the globus-sdk's v1 tokenstorage which provides simplified
+    capabilities specific to the CLI.
+    """
+
+    def __init__(self) -> None:
+        self._adapter: SQLiteAdapter | None = self._construct_adapter()
+
+    def _construct_adapter(self) -> SQLiteAdapter:
+        from globus_sdk.tokenstorage import SQLiteAdapter  # noqa: F811
+
+        # when initializing the token storage adapter, check if the storage file exists
+        # if it does not, then use this as a flag to clean the old config
+        fname = _get_storage_filename()
+        if not os.path.exists(fname):
+            from ._old_config import invalidate_old_config
```

Comment:
Did the same eyeballing with local testing here and it's in the single digit milliseconds added to run this import.

Reply:
I'll try moving this, but I note that the module in question loads `GLOBUS_SDK_ENVIRONMENT` at import time and might not respond well in the testsuite to the change.

If it works, cool. If not, I'll revert and comment back here with details.

Reply:
It worked just fine. I cannot recall why we did it that way. Maybe it was significant back when. :shrug: